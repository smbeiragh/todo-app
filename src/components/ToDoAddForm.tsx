import * as React from 'react';
import classnames from 'classnames';

type Props = {
  onSubmit: (e: any) => void
  className: string
}

type DefaultProps = {
  onSubmit: (e: any) => void
  className: string
}

export class ToDoAddForm extends React.Component<Props> {

  static defaultProps: DefaultProps = {
    className: '',
    onSubmit: ()=>{}
  }

  constructor(props: Props) {
    super(props);
    this.state = { isInvalid: false, value: '' };
  }

  handelChange(newValue: string){
    this.setState({
      isInvalid: false,
      value: newValue
    });
  }

  handleSubmit() {
    this.validate(() => {
      const { value, isInvalid } = this.state as any;
      const { onSubmit } = this.props as any;
      if(!isInvalid) {
        onSubmit({value})
        this.setState({ isInvalid: false, value: '' });
      }
    });
  }

  validate(cb: any) {
    const { value } = this.state as any;
    if (value && value.trim()) {
      this.setState({ isInvalid: false }, cb);
    } else {
      this.setState({ isInvalid: true }, cb);
    }
  }

  render() {
    const { value, isInvalid } = this.state as any;
    return (
      <div className={classnames(this.props.className, 'c-toDoAddForm', {['is-invalid']: isInvalid})} >
        <form onSubmit={(e) => { e.preventDefault(); this.handleSubmit() }}>
          <input
            placeholder="Enter Todo Name and press Enter"
            className="c-toDoAddForm__input"
            type="text"
            maxLength={300}
            value={value}
            onChange={(e)=> this.handelChange(e.target.value) }
          />
        </form>
      </div>
    );
  }
}
