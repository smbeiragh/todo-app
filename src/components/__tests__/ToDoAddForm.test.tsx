import * as React from 'react';
import { ToDoAddForm } from '../ToDoAddForm';
import * as renderer from 'react-test-renderer';

test('ToDoAddForm should render', () => {
  const component = renderer.create(
    <ToDoAddForm />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});
