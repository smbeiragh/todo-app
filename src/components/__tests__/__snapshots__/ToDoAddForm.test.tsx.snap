// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoAddForm should render 1`] = `
<div
  className="c-toDoAddForm"
>
  <form
    onSubmit={[Function]}
  >
    <input
      className="c-toDoAddForm__input"
      maxLength={300}
      onChange={[Function]}
      placeholder="Enter Todo Name and press Enter"
      type="text"
      value=""
    />
  </form>
</div>
`;
